lib/Parse/ParseStmt.cpp

```diff
@@ -1274,13 +1297,28 @@ ParserResult<PoundAvailableInfo> Parser::parseStmtConditionPoundAvailable() {
                          diag::avail_query_expected_rparen, LParenLoc))
     Status.setIsParseError();
 
+  // Diagnose #available == false as being a wrong spelling of #unavailable.
+  if (!isUnavailability && 
+      Tok.isAnyOperator() && 
+      Tok.getText() == "==" &&
+      peekToken().is(tok::kw_false)) {
+    diagnose(Tok, diag::false_available_is_called_unavailable)
+      .highlight(SourceRange(PoundLoc, peekToken().getLoc()))
+      .fixItReplace(PoundLoc, "#unavailable")
```

Comment:
Should this replacement be `!#unavailable`? It seems a bit strange for `#available` to be replaced by `#unavailable` since they have roughly the opposite meaning.

Reply:
Sorry, I misunderstood the fix-it, the `== false` is getting removed as well. This seems correct.